### PR TITLE
add cDIIS acceleration algorithm

### PR DESCRIPTION
**Pull request by tynsh**
This adds support for the cDIIS acceleration algorithm.
Some functions are renamed to use Julia conventions.

Should I commit the Manifest file separately before you merge the changes?

### CODE REVIEW

**Comment by tynsh** [comment]
I forgot to change tabs to spaces. Will do that tomorrow, so please don't merge this yet.

**Comment by tynsh** [comment]
RHF works properly, but UHF do not converge for some reason. I'm out ouf Ideas.

File: src/algorithms/cDIIS.jl [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    fock::CircularBuffer
+    error::CircularBuffer
+    errorOverlaps::CircularBuffer
+end
+
+"""
+    cDIIS
+"""
+mutable struct cDIIS <: Accelerator
+    n_prev_steps::Int
```
Comment (mfherbst) [inline src/algorithms/cDIIS.jl]: Use `n_diis_size` here

File: src/algorithms/cDIIS.jl [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        new(n_prev_steps, stateα, stateβ)
+    end
+end
+
+"""
+    Helper function.
+    pushes current fock and error matrices to states of both spin types
+"""
+function pushIterateToState!(iterate::ScfIterState, acc::cDIIS)
+    pushfirst!(acc.stateα.fock, iterate.fock[:,:,1])
```
Comment (mfherbst) [inline src/algorithms/cDIIS.jl]: Better use `view` than slices?

File: src/algorithms/cDIIS.jl [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    pushfirst!(acc.stateα.error, iterate.error[:,:,1])
+    if size(iterate.fock, 3) == 2
+        pushfirst!(acc.stateβ.fock, iterate.fock[:,:,2])
+        pushfirst!(acc.stateβ.error, iterate.error[:,:,2])
+    else
+        acc.stateβ = acc.stateα
```
Comment (mfherbst) [inline src/algorithms/cDIIS.jl]: I am not too happy that `stateβ` is dragged along even if no `β`-block is present in error or fock.

File: src/algorithms/cDIIS.jl [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+function computeNextIterate(acc::cDIIS, iterate::ScfIterState)
+    @assert acc.stateα.fock.length == acc.stateα.error.length == acc.stateβ.fock.length == acc.stateβ.error.length
+
+    # Store the current fock and error matrices first
+    pushIterateToState!(iterate, acc)
+
+    # Calculate the new fock matrix for each spin type separately
+    # and write them in the result matrix directly to save memory
+    newIterateFock = zeros(size(iterate.fock))
+    computeNextIterateFock!(acc.stateα, acc.n_prev_steps, newIterateFock[:,:,1])
```
Comment (mfherbst) [inline src/algorithms/cDIIS.jl]: Use `view` here?

**Pull request merged**